File: app/src/main/java/com/warchaser/musicplayer/tools/FormatHelper.java
package com.warchaser.musicplayer.tools;

/**
 * Created by Administrator on 2014/10/21.
 */
public class FormatHelper {
    public static String formatDuration(int iMillisecond) {
        int iSecond = iMillisecond / 1000;
        int iSecondPart = iSecond % 60;
        int iMinutePart = iSecond / 60;
        return (iMinutePart >= 10 ? iMinutePart : "0" + iMinutePart) + ":" + (iSecondPart >= 10 ? iSecondPart : "0" + iSecondPart);
    }

    public static String formatTitle(String sTitle, int iLength) {
        int iLen = sTitle.length() < iLength ? sTitle.length() : iLength;
        String sSubString = sTitle.substring(0, iLen);
        if (iLen < sTitle.length()) {
            sSubString += "...";
        }
        return sSubString;
    }

}

File: app/src/main/java/com/warchaser/musicplayer/tools/FormatHelper.kt
package com.warchaser.musicplayer.tools

/**
 * Created by Administrator on 2014/10/21.
 */
object FormatHelper {

    fun formatDuration(milliSecond : Int) : String{
        val second : Int = milliSecond / 1000
        val secondPart : Int = second % 60
        val minutePart : Int = second / 60
        val minutePartStr = if(minutePart >= 10) minutePart else "0$minutePart"
        val secondPartStr = if(secondPart >= 10) secondPart else "0$secondPart"
        return "$minutePartStr:$secondPartStr"
    }

    fun formatTitle(title : String, length : Int) : String{
        val len = if(title.length < length) title.length else length
        var subString = title.substring(0, len)
        if(len < title.length){
            subString += "..."
        }
        return subString
    }

}